src/lib/pgp-key.cpp

```diff
@@ -234,67 +234,59 @@ pgp_pk_alg_capabilities(pgp_pubkey_alg_t alg)
     }
 }
 
-static bool
-pgp_write_seckey(pgp_dest_t *   dst,
-                 pgp_pkt_type_t tag,
-                 pgp_key_pkt_t *seckey,
-                 const char *   password)
+bool
+pgp_key_t::write_sec_pgp(pgp_dest_t &dst, pgp_key_pkt_t &seckey, const std::string &password)
 {
     bool           res = false;
-    pgp_pkt_type_t oldtag = seckey->tag;
+    pgp_pkt_type_t oldtag = seckey.tag;
 
-    seckey->tag = tag;
-    if (encrypt_secret_key(seckey, password, NULL)) {
+    seckey.tag = type();
+    if (encrypt_secret_key(&seckey, password.c_str(), NULL)) {
         goto done;
     }
     try {
-        seckey->write(*dst);
-        res = !dst->werr;
+        seckey.write(dst);
+        res = !dst.werr;
     } catch (const std::exception &e) {
         RNP_LOG("%s", e.what());
     }
 done:
-    seckey->tag = oldtag;
+    seckey.tag = oldtag;
     return res;
 }
 
-static bool
-write_key_to_rawpacket(pgp_key_pkt_t *        seckey,
-                       pgp_rawpacket_t &      packet,
-                       pgp_pkt_type_t         type,
-                       pgp_key_store_format_t format,
-                       const char *           password)
+bool
+pgp_key_t::write_sec_rawpkt(pgp_key_pkt_t &seckey, const std::string &password)
 {
     pgp_dest_t memdst = {};
-    bool       ret = false;
-
     if (init_mem_dest(&memdst, NULL, 0)) {
-        goto done;
+        return false;
     }
 
+    bool ret = false;
     // encrypt+write the key in the appropriate format
-    switch (format) {
-    case PGP_KEY_STORE_GPG:
-    case PGP_KEY_STORE_KBX:
-        if (!pgp_write_seckey(&memdst, type, seckey, password)) {
-            RNP_LOG("failed to write seckey");
-            goto done;
-        }
-        break;
-    case PGP_KEY_STORE_G10:
-        if (!g10_write_seckey(&memdst, seckey, password)) {
-            RNP_LOG("failed to write g10 seckey");
+    try {
+        switch (format) {
+        case PGP_KEY_STORE_GPG:
+        case PGP_KEY_STORE_KBX:
+            if (!write_sec_pgp(memdst, seckey, password)) {
+                RNP_LOG("failed to write seckey");
+                goto done;
+            }
+            break;
+        case PGP_KEY_STORE_G10:
+            if (!g10_write_seckey(&memdst, &seckey, password.c_str())) {
+                RNP_LOG("failed to write g10 seckey");
```

Comment:
"seckey"

Reply:
Actually this message and previous one were not changed by this PR, but let's make things more understandable, will force-push shortly.